pulumi/infra/swarm.py

```diff
@@ -24,28 +24,46 @@ class Ec2Port:
 
     def allow_internally(
         self,
-    ) -> Tuple[aws.ec2.SecurityGroupIngressArgs, aws.ec2.SecurityGroupEgressArgs]:
-        ingress = aws.ec2.SecurityGroupIngressArgs(
+        sg: aws.ec2.SecurityGroup,
+    ) -> Tuple[aws.ec2.SecurityGroupRule, aws.ec2.SecurityGroupRule]:
+        ingress = aws.ec2.SecurityGroupRule(
+            f"internal-ingress-{self}",
+            type="ingress",
+            security_group_id=sg.id,
             from_port=self.port,
             to_port=self.port,
             protocol=self.protocol,
             self=True,
+            opts=pulumi.ResourceOptions(parent=sg),
         )
-        egress = aws.ec2.SecurityGroupEgressArgs(
+
+        egress = aws.ec2.SecurityGroupRule(
+            f"internal-egress-{self}",
+            type="egress",
+            security_group_id=sg.id,
             from_port=self.port,
             to_port=self.port,
             protocol=self.protocol,
             self=True,
+            opts=pulumi.ResourceOptions(parent=sg),
         )
+
         return (ingress, egress)
 
-    def allow_outbound_any_ip(self) -> aws.ec2.SecurityGroupEgressArgs:
-        return aws.ec2.SecurityGroupEgressArgs(
+    def allow_outbound_any_ip(
```

Comment:
We don't actually need to return anything from this function, or from `allow_internally`